climpred/classes.py

```diff
@@ -250,15 +434,26 @@ def compute_metric(self, metric='pearson_r', comparison='m2m'):
         Returns:
             Result of the comparison as a Dataset.
         """
-        is_initialized(self.control, 'control', 'predictability')
-        return compute_perfect_model(
-            self.initialized, self.control, metric=metric, comparison=comparison
+        is_initialized(self._datasets['control'], 'control', 'predictability')
+        input_dict = {
+            'ensemble': self._datasets['initialized'],
+            'control': self._datasets['control'],
+            'var': var,
+            'init': True,
+        }
+        return self._apply_climpred_function(
+            compute_perfect_model,
+            input_dict=input_dict,
+            metric=metric,
+            comparison=comparison,
         )
 
-    def compute_uninitialized(self, metric='pearson_r', comparison='m2e'):
+    def compute_uninitialized(self, var=None, metric='pearson_r', comparison='m2e'):
```

Comment:
Didn’t we want to drop compute_uninit?

Reply:
Yes, I am going to address this in another PR so that this one doesn't creep too far from the main goal. I know it's a somewhat easy thing to do but I want to have that PR contained so I can do testing appropriately.